include/stdexec/execution.hpp

```diff
@@ -1914,6 +1930,8 @@ namespace stdexec {
         if constexpr (!enable_receiver<__decay_t<_Receiver>>)
           _PLEASE_UPDATE_YOUR_RECEIVER_TYPE< __decay_t<_Receiver>>();
 
+        static_assert(__check_signatures<__decay_t<_Sender>, env_of_t<__decay_t<_Receiver>>>());
```

Comment:
i don't think you want to be decaying the sender and receiver types here. cvref qualification sometimes matters.